File: python/docmost/services/workspace_api.py
"""Generated API client."""
from __future__ import annotations

from typing import Any, Optional

import httpx

from docmost.http_client import DocmostHttpClient
from docmost.types import ApiResponse
from docmost.models import AcceptInviteDto, CheckHostnameDto, InvitationIdDto, InviteUserDto, PaginationOptions, RemoveWorkspaceUserDto, UpdateWorkspaceDto, UpdateWorkspaceUserRoleDto


class WorkspaceApi:
    """Workspace API."""

    def __init__(self, http: DocmostHttpClient) -> None:
        self._http = http

    def get_workspace_public_info(self) -> ApiResponse[Any]:
        return self._http.post("workspace/public", body=None)

    def get_workspace_info(self) -> ApiResponse[Any]:
        return self._http.post("workspace/info", body=None)

    def update_workspace(self, request: UpdateWorkspaceDto) -> ApiResponse[Any]:
        return self._http.post("workspace/update", body=request.model_dump(by_alias=True, exclude_none=True))

    def get_workspace_members(self, request: PaginationOptions) -> ApiResponse[Any]:
        return self._http.post("workspace/members", body=request.model_dump(by_alias=True, exclude_none=True))

    def deactivate_workspace_member(self) -> ApiResponse[Any]:
        return self._http.post("workspace/members/deactivate", body=None)

    def delete_workspace_member(self, request: RemoveWorkspaceUserDto) -> ApiResponse[Any]:
        return self._http.post("workspace/members/delete", body=request.model_dump(by_alias=True, exclude_none=True))

    def update_workspace_member_role(self, request: UpdateWorkspaceUserRoleDto) -> ApiResponse[Any]:
        return self._http.post("workspace/members/change-role", body=request.model_dump(by_alias=True, exclude_none=True))

    def get_invitations(self, request: PaginationOptions) -> ApiResponse[Any]:
        return self._http.post("workspace/invites", body=request.model_dump(by_alias=True, exclude_none=True))

    def get_invitation_by_id(self, request: InvitationIdDto) -> ApiResponse[Any]:
        return self._http.post("workspace/invites/info", body=request.model_dump(by_alias=True, exclude_none=True))

    def invite_user(self, request: InviteUserDto) -> ApiResponse[Any]:
        return self._http.post("workspace/invites/create", body=request.model_dump(by_alias=True, exclude_none=True))

    def resend_invite(self, request: InvitationIdDto) -> ApiResponse[Any]:
        return self._http.post("workspace/invites/resend", body=request.model_dump(by_alias=True, exclude_none=True))

    def revoke_invite(self, request: InvitationIdDto) -> ApiResponse[Any]:
        return self._http.post("workspace/invites/revoke", body=request.model_dump(by_alias=True, exclude_none=True))

    def accept_invite(self, request: AcceptInviteDto) -> ApiResponse[Any]:
        return self._http.post("workspace/invites/accept", body=request.model_dump(by_alias=True, exclude_none=True))

    def get_invite_link(self, request: InvitationIdDto) -> ApiResponse[Any]:
        return self._http.post("workspace/invites/link", body=request.model_dump(by_alias=True, exclude_none=True))

    def check_hostname(self, request: CheckHostnameDto) -> ApiResponse[Any]:
        return self._http.post("workspace/check-hostname", body=request.model_dump(by_alias=True, exclude_none=True))

File: go/docmost/services/workspace_api.go
package services

import (
	"context"

	"github.com/ggpo1/docmost-sdk/go/internal/httpclient"
	"github.com/ggpo1/docmost-sdk/go/docmost/models"
)

// WorkspaceAPI implements Workspace API.
type WorkspaceAPI struct {
	http *httpclient.HTTPClient
}

func NewWorkspaceAPI(http *httpclient.HTTPClient) *WorkspaceAPI {
	return &WorkspaceAPI{http: http}
}

func (a *WorkspaceAPI) GetWorkspacePublicInfo(ctx context.Context) (*httpclient.ApiResponse, error) {
	return a.http.Post(ctx, "workspace/public", nil)
}

func (a *WorkspaceAPI) GetWorkspaceInfo(ctx context.Context) (*httpclient.ApiResponse, error) {
	return a.http.Post(ctx, "workspace/info", nil)
}

func (a *WorkspaceAPI) UpdateWorkspace(ctx context.Context, request *models.UpdateWorkspaceDto) (*httpclient.ApiResponse, error) {
	return a.http.Post(ctx, "workspace/update", request)
}

func (a *WorkspaceAPI) GetWorkspaceMembers(ctx context.Context, request *models.PaginationOptions) (*httpclient.ApiResponse, error) {
	return a.http.Post(ctx, "workspace/members", request)
}

func (a *WorkspaceAPI) DeactivateWorkspaceMember(ctx context.Context) (*httpclient.ApiResponse, error) {
	return a.http.Post(ctx, "workspace/members/deactivate", nil)
}

func (a *WorkspaceAPI) DeleteWorkspaceMember(ctx context.Context, request *models.RemoveWorkspaceUserDto) (*httpclient.ApiResponse, error) {
	return a.http.Post(ctx, "workspace/members/delete", request)
}

func (a *WorkspaceAPI) UpdateWorkspaceMemberRole(ctx context.Context, request *models.UpdateWorkspaceUserRoleDto) (*httpclient.ApiResponse, error) {
	return a.http.Post(ctx, "workspace/members/change-role", request)
}

func (a *WorkspaceAPI) GetInvitations(ctx context.Context, request *models.PaginationOptions) (*httpclient.ApiResponse, error) {
	return a.http.Post(ctx, "workspace/invites", request)
}

func (a *WorkspaceAPI) GetInvitationById(ctx context.Context, request *models.InvitationIdDto) (*httpclient.ApiResponse, error) {
	return a.http.Post(ctx, "workspace/invites/info", request)
}

func (a *WorkspaceAPI) InviteUser(ctx context.Context, request *models.InviteUserDto) (*httpclient.ApiResponse, error) {
	return a.http.Post(ctx, "workspace/invites/create", request)
}

func (a *WorkspaceAPI) ResendInvite(ctx context.Context, request *models.InvitationIdDto) (*httpclient.ApiResponse, error) {
	return a.http.Post(ctx, "workspace/invites/resend", request)
}

func (a *WorkspaceAPI) RevokeInvite(ctx context.Context, request *models.InvitationIdDto) (*httpclient.ApiResponse, error) {
	return a.http.Post(ctx, "workspace/invites/revoke", request)
}

func (a *WorkspaceAPI) AcceptInvite(ctx context.Context, request *models.AcceptInviteDto) (*httpclient.ApiResponse, error) {
	return a.http.Post(ctx, "workspace/invites/accept", request)
}

func (a *WorkspaceAPI) GetInviteLink(ctx context.Context, request *models.InvitationIdDto) (*httpclient.ApiResponse, error) {
	return a.http.Post(ctx, "workspace/invites/link", request)
}

func (a *WorkspaceAPI) CheckHostname(ctx context.Context, request *models.CheckHostnameDto) (*httpclient.ApiResponse, error) {
	return a.http.Post(ctx, "workspace/check-hostname", request)
}

